octobot_trading/data/order.py

```diff
@@ -271,60 +271,36 @@ def update_from_raw(self, raw_order):
-            "timestamp": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.TIMESTAMP.value, None)
-        })
-
-    @staticmethod
-    def parse_order_type(raw_order):
-        side: TradeOrderSide = TradeOrderSide(raw_order[ExchangeConstantsOrderColumns.SIDE.value])
-        order_type: TradeOrderType = TradeOrderType(raw_order[ExchangeConstantsOrderColumns.TYPE.value])
-
-        if side == TradeOrderSide.BUY:
-            if order_type == TradeOrderType.LIMIT:
-                order_type = TraderOrderType.BUY_LIMIT
-            elif order_type == TradeOrderType.MARKET:
-                order_type = TraderOrderType.BUY_MARKET
-        elif side == TradeOrderSide.SELL:
-            if order_type == TradeOrderType.LIMIT:
-                order_type = TraderOrderType.SELL_LIMIT
-            elif order_type == TradeOrderType.MARKET:
-                order_type = TraderOrderType.SELL_MARKET
-        return side, order_type
-
-    @staticmethod
-    def parse_order_status(raw_order):
-        try:
-            return OrderStatus(raw_order[ExchangeConstantsOrderColumns.STATUS.value])
-        except KeyError:
-            return None
+        return self.update(
+            symbol=str(get_value_or_default(raw_order, ExchangeConstantsOrderColumns.SYMBOL.value, None)),
+            current_price=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.PRICE.value, 0.0),
+            quantity=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.AMOUNT.value, 0.0),
+            price=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.PRICE.value, 0.0),
+            status=parse_order_status(raw_order),
+            order_id=str(get_value_or_default(raw_order, ExchangeConstantsOrderColumns.ID.value, None)),
+            quantity_filled=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.FILLED.value, 0.0),
+            filled_price=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.PRICE.value, 0.0),
+            total_cost=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.COST.value, 0.0),
+            fee=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.FEE.value, None),
+            timestamp=get_value_or_default(raw_order, ExchangeConstantsOrderColumns.TIMESTAMP.value, None)
+        )
 
     def update_order_from_raw(self, raw_order):
-        self.status = Order.parse_order_status(raw_order)
+        self.status = parse_order_status(raw_order)
         self.total_cost = raw_order[ExchangeConstantsOrderColumns.COST.value]
         self.filled_quantity = raw_order[ExchangeConstantsOrderColumns.FILLED.value]
         self.filled_price = raw_order[ExchangeConstantsOrderColumns.PRICE.value]
         if not self.filled_price and self.filled_quantity:
             self.filled_price = self.total_cost / self.filled_quantity
-        self.taker_or_maker = Order.parse_order_type(raw_order)
+
+        self.taker_or_maker = parse_order_type(raw_order)
```

Comment:
Is parse_order_type(raw_order) telling if the order is a taker or a maker ?

Reply:
Oh no you're right, I don't know where this line come from.